pkg/analyzer/lfp_npm.go

```diff
@@ -284,6 +284,17 @@ func npmIsUrlFollowsPathConvention(sourceUrl string, pkg string, trustedUrls []s
 
 	// Example: @angular/core from https://registry.npmjs.org/@angular/core/-/core-1.0.0.tgz
 	scopedPackageName := strings.Split(path, "/-/")[0]
+	if slices.Contains(acceptablePackageNames, scopedPackageName) {
+		return true
+	} else {
+		// Check if resolved URL starts with any trusted URL
+		// This is used to handle cases where strip-ansi-cjs resolves to https://registry.npmjs.org/strip-ansi/-/strip-ansi-6.0.1.tgz
+		for _, trustedUrl := range trustedUrls {
+			if strings.HasPrefix(sourceUrl, trustedUrl) {
```

Comment:
Don't think this is correct. For example `https://registry.npmjs.org` is a trusted URL. Within this, everything will match including `https://registry.npmjs.org/x/y/z` which may not be desirable. For example, this E2E test case caught the problem:

https://github.com/safedep/vet/blob/main/test/scenarios/fixtures/lockfiles/nestjs-lfp-package-lock.json#L68

May be we need to do this check only with `npm.config.TrustedRegistryUrls` that are explicitly supplied by the users.